### PR TITLE
Base on graded assignment toggle

### PR DESCRIPTION
**Pull request by szabinst**
Base on graded assignment toggle.

- Added core functionality 
- Locked state 
- Loading state when switching grading periods

refs: MBL17302
affects: Student, Parent
release note: Added the ability to show grades based on graded assignments
test plan: Compare to Android, Web

## Screenshots

<table>
<tr><th>Toggle</th><th>Locked state</th><th>Loading state</th></tr>
<tr>
<td></td>
<td></td>
<td></td>
</tr>
</table>

## Checklist

- [ ] Follow-up e2e test ticket created
- [x] A11y checked
- [x] Tested on phone
- [x] Tested on tablet
- [x] Tested in dark mode
- [x] Tested in light mode

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Review by vargaat — CHANGES_REQUESTED** [review]
Looks nice, this is a long awaited feature. A few things to improve:
- The letter grade doesn't seem to update.
- The score update when switching the value lags, guess because of the throttle, any ways to avoid that? Not a big thing if not.
- On iOS 16 the switch's state isn't announced at all via VoiceOver. On iOS 17 it works as expected.
- This screen is shared with the parent app so let's add that as affected.

**Review by vargaat — DISMISSED** [review]
QA+1, worked well with my test accounts.

File: Core/Core/Grades/View/GradeListView.swift [hunk 1]
```diff
@@ -185,64 +202,98 @@ public struct GradeListView: View, ScreenViewTrackable {
...
+//            Toggle(isOn: $viewModel.isWhatIfScoreOn) {
+//                Text("Show What-if Score", bundle: .core)
+//                    .foregroundStyle(Color.textDarkest)
+//                    .font(.regular16)
+//                    .fixedSize()
+//                    .lineLimit(1)
+//            }
+//            .toggleStyle(SwitchToggleStyle(tint: Color(Brand.shared.primary)))
+//            .frame(height: 51)
+//            .padding(.horizontal, 16)
```
Comment (rh12) [inline Core/Core/Grades/View/GradeListView.swift:296]: Just to doublecheck: is the commented out part intentional?

File: Core/Core/Grades/Model/GradeListInteractor.swift [hunk 2]
```diff
@@ -311,10 +339,19 @@ public final class GradeListInteractorLive: GradeListInteractor {
...
+                } else {
+                    localGrade = gradeEnrollment?.formattedFinalScore(gradingPeriodID: gradingPeriodID)
+                    letterGrade = gradeEnrollment?.finalGrade(gradingPeriodID: gradingPeriodID)
+                }
             } else {
-                localGrade = courseEnrollment?.formattedCurrentScore(gradingPeriodID: nil)
+                if baseOnGradedAssignments {
+                    localGrade = courseEnrollment?.formattedCurrentScore(gradingPeriodID: nil)
+                } else {
+                    localGrade = gradeEnrollment?.formattedFinalScore(gradingPeriodID: nil)
```
[the remark this answers is not in the document]
Comment (szabinst) [reply Core/Core/Grades/Model/GradeListInteractor.swift:353]: In this case, it shouldn't.

File: Core/Core/Grades/ViewModel/GradeListViewModel.swift [hunk 3]
```diff
@@ -65,35 +73,55 @@ public final class GradeListViewModel: ObservableObject {
     ) {
         self.interactor = interactor
 
-        let triggerRefresh = PassthroughRelay<(Bool, RefreshCompletion)>()
+        let triggerRefresh = PassthroughRelay<(IgnoreCache, RefreshCompletion)>()
```
[the remark this answers is not in the document]
Comment (szabinst) [reply Core/Core/Grades/ViewModel/GradeListViewModel.swift]: Good suggestion, changed it.

File: Core/Core/Grades/View/GradeListView.swift (hunk 1 above)
Comment (szabinst) [reply Core/Core/Grades/View/GradeListView.swift:297]: Yes, that's the What-if toggle that is not functional yet.

**Review by vargaat — APPROVED** [review]

**Review by rh12 — APPROVED** [review]
QA+1
Thanks for the changes!

**Pull request merged**